src/index.ts

```diff
@@ -33,28 +34,36 @@ namespace CommandIDs {
  * A service providing variable introspection.
  */
 const variableinspector: JupyterFrontEndPlugin<IVariableInspectorManager> = {
-  id: '@lckr/jupyterlab_variableinspector',
-  requires: [ICommandPalette, ILayoutRestorer, ILabShell],
+  id: '@lckr/jupyterlab_variableinspector:plugin',
+  requires: [ICommandPalette, ILayoutRestorer, ILabShell, ISettingRegistry],
   provides: IVariableInspectorManager,
   autoStart: true,
-  activate: (
+  activate: async (
     app: JupyterFrontEnd,
     palette: ICommandPalette,
     restorer: ILayoutRestorer,
-    labShell: ILabShell
-  ): IVariableInspectorManager => {
+    labShell: ILabShell,
+    settingRegistry: ISettingRegistry
+  ): Promise<IVariableInspectorManager> => {
     const manager = new VariableInspectorManager();
     const category = 'Variable Inspector';
     const command = CommandIDs.open;
     const label = 'Open Variable Inspector';
     const namespace = 'variableinspector';
     const tracker = new WidgetTracker<VariableInspectorPanel>({ namespace });
+    let settings: ISettingRegistry.ISettings;
+
+    try {
+      settings = await settingRegistry.load(variableinspector.id);
+    } catch (error) {
+      console.error('Failed to load settings for the Git Extension', error);
+    }
 
     /**
      * Create and track a new inspector.
      */
     function newPanel(): VariableInspectorPanel {
-      const panel = new VariableInspectorPanel();
+      const panel = new VariableInspectorPanel(settings);
```

Comment:
This is not the best approach as it add an opinionated interface on how the filter state is stored. To ease code maintenance, extensibility and testing, the best is to not passed the settings object but rather to add filter setters on the panel.

Now the trouble raises from how to pass information from the `settings` only available within the promise to the panel.

> The best solution would be to refactor substantially the code as it is a bad idea to embed the view `panel` in the `manager`.

But let's go for a simpler fix for now. You could use a `PromiseDelegate` like this:

```ts
const settings = new PromiseDelegate<ISettingRegistry.ISettings>();

Promise.all([settingRegistry.load(variableinspector.id), app.restored])
.then(settings_ => {
  settings.resolve(settings_);
})
.catch(error => {
  console.error('Failed to load settings for the Git Extension', error);
  settings.reject(error);
});
```

Then change the command `execute` to be async:

```ts
      execute: async () => {
        if (!manager.panel || manager.panel.isDisposed) {
          await settings; // Wait for the settings to be loaded
          manager.panel = newPanel();
          // Best would be to use a single array of filter object: {filter: '', type: ''}[]
          // to clarify the binding between the type and the filter
          manager.panel.filters = settings.filters;
        }
        if (!manager.panel.isAttached) {
          labShell.add(manager.panel, 'main');
        }
        if (manager.source) {
          manager.source.performInspection();
        }
        labShell.activateById(manager.panel.id);
      }
```